### PR TITLE
Make completions a HLS plugin

### PR DESCRIPTION
**Pull request by gdevanla**
This replaces PR #670 and partially addresses #724

The end goal of this PR would be to abstract away the completions code (both Local and NonLocal) into one HLS plugin or separate HLS plugins one for each.

### CODE REVIEW

**Comment by pepeiborra** [comment]
There should be no need to move the Local Completions code around anymore. I would prefer to have a diff as small as possible

**Comment by gdevanla** [comment]
I see. So, do we still want to break up Completion.hs into multiple HLS plugins' are should we not be doing that as well?  

In other words, do we want to have Logic.hs intact and see if can extract away Completion.hs into multiple HLS plugins?

File: ghcide/src/Development/IDE/Plugin/HLS/Completions.hs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+descriptor :: PluginId -> PluginDescriptor IdeState
+descriptor plId = (defaultPluginDescriptor plId)
+  { pluginCommands = []
+  , pluginCodeActionProvider = Nothing
+  , pluginCodeLensProvider   = Nothing
+  , pluginHoverProvider      = Nothing
+  , pluginSymbolsProvider    = Nothing
+  , pluginCompletionProvider = Just getCompletionsLSP
+  , pluginRules              = Ghcide.pluginRules Completions.plugin
+  }
```
Comment (gdevanla) [inline ghcide/src/Development/IDE/Plugin/HLS/Completions.hs:30]: Alternatively, we can move code in Plugins/Completions.hs into this module and get rid of that module.  That will reduce this redirection which seems redudant.

Any thoughts?

**Comment by pepeiborra** [comment]
Either solution that allows to extract HLS providers to minimise the diff and preserve the git history. Probably keep things where they are and just break them down in smaller functions which can be assembled into an HLS provider in Plugin.HLS.Completions

**Comment by gdevanla** [comment]
I think I will abandon this effort for now. The recent move seems to achieve what we want. A moving around code may not bring much material benefits.

**Pull request closed without merging**